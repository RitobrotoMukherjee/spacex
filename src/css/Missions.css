.mission-container {
  padding: 15px;
}

.loading {
  color: green;
  font-size: larger;
  font-weight: 600;
}

.missionTable {
  border-collapse: collapse;
  margin: 121px auto;
  table-layout: auto;
  width: 90%;
}

.table-heading {
  font-size: 1.3rem;
  padding: 10px;
}

.row:nth-child(odd) {
  background-color: #e8e4e4;
}

.column {
  border: 1px solid rgb(165, 164, 164);
  padding: 10px 10px 1.8rem;
  text-align: left;
}

.mission-name {
  font-size: 1.3rem;
  font-weight: 700;
  vertical-align: top;
}

.mission-desc {
  color: #474a4e;
}

.mission-status {
  padding: 0 1rem;
}

.mission-button {
  padding: 0 3rem;
}

.reserved,
.not-reserved {
  border-radius: 6px;
  color: #fff;
  font-weight: 700;
  line-height: 24px;
  margin: 0;
  min-width: 140px;
  padding: 1px 6px 0;
  text-align: center;
}

.reserved {
  background: #20b9df;
}

.not-reserved {
  background: #6f7275;
}

td button {
  background: transparent;
  border-radius: 5px;
  cursor: pointer;
  font-size: 1.3rem;
  min-width: 150px;
}

.btn {
  border-color: #444a4f;
  color: #444a4f;
  line-height: 40px;
}

.btn-leave {
  border-color: #f0080c;
  color: #f0080c;
}
