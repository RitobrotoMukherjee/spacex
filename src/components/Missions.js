import '../css/Missions.css';
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissions } from '../redux/missions/missions';
import MissionTable from './Mission-components/MissionsTable';

const Missions = () => {
  const dispatch = useDispatch();

  const { loading, missionList } = useSelector((state) => state.missions);

  useEffect(() => {
    if (missionList.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch]);

  return (
    <section className="mission-container">
      {loading
        && (<h2 className="loading">Loading...</h2>)}
      {!loading && (<MissionTable />)}
    </section>
  );
};

export default Missions;
