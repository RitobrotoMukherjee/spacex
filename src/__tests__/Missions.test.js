import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Missions from '../components/Missions';

it('Missions component snapshot testing', () => {
  const missions = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Missions />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(missions).toMatchSnapshot();
});
