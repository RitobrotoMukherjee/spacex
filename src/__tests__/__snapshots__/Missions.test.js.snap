// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component snapshot testing 1`] = `
<section
  className="mission-container"
>
  <h2
    className="loading"
  >
    Loading...
  </h2>
</section>
`;
